### ISSUE TITLE
Exception in nozzle calibration

### ISSUE THREAD

**Issue by JulienDevillers** [openpnp/openpnp#826]
# Problem Report
## Describe the Issue
I try to calibrate a nozzle. After 2 rotations, one in each direction, the functions fails with a dialog box "java.lang.exception cannot be cast to java.util.List"
Version  2019_02_09_19-58-57 296d901 .

## Steps to Reproduce
How can a developer reproduce the issue you are experiencing? Use a step by step format like this:
1. A machine with a nozzle and a nozzle tip.
2. navigate to nozzle calbration panel
3. move the nozzle above the camera
4. Set the pipeline :
```
<cv-pipeline>
   <stages>
      <cv-stage class="org.openpnp.vision.pipeline.stages.ImageCapture" name="9" enabled="true" settle-first="true"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.ConvertColor" name="1" enabled="true" conversion="Bgr2Gray"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.BlurMedian" name="4" enabled="true" kernel-size="3"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.MaskCircle" name="3" enabled="true" diameter="200"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.Threshold" name="2" enabled="true" threshold="150" auto="false" invert="false"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.MaskCircle" name="5" enabled="false" diameter="100"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.DetectCirclesHough" name="6" enabled="true" min-distance="50" min-diameter="40" max-diameter="60" dp="1.0" param-1="80.0" param-2="10.0"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.SimpleBlobDetector" name="results" enabled="false" threshold-step="10.0" threshold-min="50.0" threshold-max="220.0" repeatability="2" dist-between-blobs="10.0" color="true" color-value="0.0" area="true" area-min="25.0" area-max="5000.0" circularity="false" circularity-min="0.800000011920929" circularity-max="-1.0" inertia="true" inertia-ratio-min="0.10000000149011612" inertia-ratio-max="-1.0" convexity="true" convexity-min="0.949999988079071" convexity-max="-1.0"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.ImageRecall" name="0" enabled="true" image-stage-name="9"/>
      <cv-stage class="org.openpnp.vision.pipeline.stages.DrawKeyPoints" name="7" enabled="false" key-points-stage-name="results">
         <color r="255" g="255" b="0" a="255"/>
      </cv-stage>
      <cv-stage class="org.openpnp.vision.pipeline.stages.DrawCircles" name="8" enabled="true" circles-stage-name="6" thickness="3"/>
   </stages>
</cv-pipeline>
```

4. click "calibrate" button

log:
```
2019-05-08 12:53:06.568 Scripting TRACE: Scripting.on NozzleCalibration.Starting
2019-05-08 12:53:06.568 ReferenceHead DEBUG: H1.moveToSafeZ(0.9)
2019-05-08 12:53:06.568 ReferenceNozzle DEBUG: N1.moveToSafeZ(0.9)
2019-05-08 12:53:06.568 ReferenceCamera DEBUG: Head.moveToSafeZ(0.9)
2019-05-08 12:53:06.568 ReferenceActuator DEBUG: A1.moveToSafeZ(0.9)
2019-05-08 12:53:06.569 ReferenceNozzle DEBUG: N1.moveTo((-74.000000, 24.800000, -3.000000, 0.000000 mm), 0.9)
2019-05-08 12:53:06.569 ReferenceNozzle DEBUG: N1.moveTo((-74.000000, 24.800000, -40.000000, 0.000000 mm), 0.9)
2019-05-08 12:53:06.569 GcodeDriver DEBUG: sendCommand(G0   Z-43.0000  F7200 ; Send standard Gcode move, 10000)...
2019-05-08 12:53:06.569 GcodeDriver TRACE: [serial://COM44] >> G0   Z-43.0000  F7200 ; Send standard Gcode move
2019-05-08 12:53:06.570 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:06.570 GcodeDriver DEBUG: sendCommand(serial://COM44 G0   Z-43.0000  F7200 ; Send standard Gcode move, 10000) => [ok]
2019-05-08 12:53:06.570 GcodeDriver DEBUG: sendCommand(M400 ; Wait for moves to complete before returning, 10000)...
2019-05-08 12:53:06.570 GcodeDriver TRACE: [serial://COM44] >> M400 ; Wait for moves to complete before returning
2019-05-08 12:53:07.701 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:07.701 GcodeDriver DEBUG: sendCommand(serial://COM44 M400 ; Wait for moves to complete before returning, 10000) => [ok]
2019-05-08 12:53:07.701 ReferenceNozzleTip$Calibration DEBUG: [nozzleTipCalibration]starting measurement; angleStart: -180.0, angleStop: 180.0, angleIncrement: 60.0, angleSubdivisions: 5
2019-05-08 12:53:07.701 ReferenceNozzleTip$Calibration DEBUG: [nozzleTipCalibration]i: 0, measureAngle: -180.0
2019-05-08 12:53:07.701 ReferenceNozzle DEBUG: N1.moveTo((-74.000000, 24.800000, -40.000000, -180.000000 mm), 0.9)
2019-05-08 12:53:07.701 GcodeDriver DEBUG: sendCommand(G0    E-180.0000 F7200 ; Send standard Gcode move, 10000)...
2019-05-08 12:53:07.701 GcodeDriver TRACE: [serial://COM44] >> G0    E-180.0000 F7200 ; Send standard Gcode move
2019-05-08 12:53:07.702 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:07.702 GcodeDriver DEBUG: sendCommand(serial://COM44 G0    E-180.0000 F7200 ; Send standard Gcode move, 10000) => [ok]
2019-05-08 12:53:07.702 GcodeDriver DEBUG: sendCommand(M400 ; Wait for moves to complete before returning, 10000)...
2019-05-08 12:53:07.702 GcodeDriver TRACE: [serial://COM44] >> M400 ; Wait for moves to complete before returning
2019-05-08 12:53:11.414 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:11.414 GcodeDriver DEBUG: sendCommand(serial://COM44 M400 ; Wait for moves to complete before returning, 10000) => [ok]
2019-05-08 12:53:11.414 Scripting TRACE: Scripting.on Camera.BeforeSettle
2019-05-08 12:53:11.664 Scripting TRACE: Scripting.on Camera.BeforeCapture
2019-05-08 12:53:11.667 Scripting TRACE: Scripting.on Camera.AfterCapture
2019-05-08 12:53:11.681 ReferenceNozzle DEBUG: N1.moveTo((-74.000000, 24.800000, -40.000000, 0.000000 mm), 0.9)
2019-05-08 12:53:11.682 GcodeDriver DEBUG: sendCommand(G0    E0.0000 F7200 ; Send standard Gcode move, 10000)...
2019-05-08 12:53:11.682 GcodeDriver TRACE: [serial://COM44] >> G0    E0.0000 F7200 ; Send standard Gcode move
2019-05-08 12:53:11.683 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:11.683 GcodeDriver DEBUG: sendCommand(serial://COM44 G0    E0.0000 F7200 ; Send standard Gcode move, 10000) => [ok]
2019-05-08 12:53:11.683 GcodeDriver DEBUG: sendCommand(M400 ; Wait for moves to complete before returning, 10000)...
2019-05-08 12:53:11.683 GcodeDriver TRACE: [serial://COM44] >> M400 ; Wait for moves to complete before returning
2019-05-08 12:53:15.394 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:15.394 GcodeDriver DEBUG: sendCommand(serial://COM44 M400 ; Wait for moves to complete before returning, 10000) => [ok]
2019-05-08 12:53:15.395 ReferenceNozzle DEBUG: N1.moveToSafeZ(0.9)
2019-05-08 12:53:15.395 GcodeDriver DEBUG: sendCommand(G0   Z-6.0000  F7200 ; Send standard Gcode move, 10000)...
2019-05-08 12:53:15.395 GcodeDriver TRACE: [serial://COM44] >> G0   Z-6.0000  F7200 ; Send standard Gcode move
2019-05-08 12:53:15.396 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:15.396 GcodeDriver DEBUG: sendCommand(serial://COM44 G0   Z-6.0000  F7200 ; Send standard Gcode move, 10000) => [ok]
2019-05-08 12:53:15.396 GcodeDriver DEBUG: sendCommand(M400 ; Wait for moves to complete before returning, 10000)...
2019-05-08 12:53:15.396 GcodeDriver TRACE: [serial://COM44] >> M400 ; Wait for moves to complete before returning
2019-05-08 12:53:16.526 GcodeDriver TRACE: [serial://COM44] << ok
2019-05-08 12:53:16.527 GcodeDriver DEBUG: sendCommand(serial://COM44 M400 ; Wait for moves to complete before returning, 10000) => [ok]
2019-05-08 12:53:16.527 MessageBoxes DEBUG: Error: java.lang.ClassCastException: java.lang.Exception cannot be cast to java.util.List

```

## Expected Result
No failure.

## Actual Result
Error message.

**Comment by markmaker** [comment]
There is not enough information. Please add the log. 

What version of OpenPNP do you use?

Have you ever tried to reset the Pipeline? The default pipeline was dramatically changed in recent `develop` branch versions. If you have recently updated the version, maybe the pipeline stored on your nozzle tip is no longer compatible with the code. 

_Mark

**Comment by JulienDevillers** [comment]
Ok, I added version number.
Sorry, I wrote the pipeline in the issue but as it is xml, it did not appear in the issue. I set it as code and now you can see it.

The default pipeline does not recognize the nozzle. That's why I'm using a custom one.

**Comment by JulienDevillers** [comment]
I did some debug.
In ReferenceNozzleTip.java

In my pipeline, the stage named "result" is disabled. This causes
` Object results = pipeline.getResult(VisionUtils.PIPELINE_RESULTS_NAME).model;
`
to return " java.lang.exception : Stage not enabled." which can't be casted to List :
 `              if (0==((List) results).size()) {
 `
Fix by #833

**Comment by JulienDevillers** [comment]
Fixed by #833
